modules/govuk/manifests/node/s_collections_publisher_db_admin.pp

```diff
@@ -0,0 +1,45 @@
+# == Class: govuk_node::s_collections_db_admin
+#
+# This machine class is used to administer the Collections Publisher
+# MySQL RDS instances.
+#
+# === Parameters
+#
+# [*mysql_db_host*]
+#  The database hostname
+#
+# [*mysql_db_password*]
+#  The database password
+#
+# [*mysql_db_user*]
+#  The database user to connect to the remote database as
+#
+# [*apt_mirror_hostname*]
+#   The hostname for the apt mirror to add to enable fetching specific
+#   packages
+#
+#  [*apt_mirror_gpg_key_fingerprint*]
+#    The fingerprint of the local aptly mirror.
+#
+class govuk::node::s_collections_publisher_db_admin(
+  $mysql_db_host        = undef,
+  $mysql_db_password    = undef,
+  $mysql_db_user        = undef,
+  $mysql_backup_hour    = 9,
+  $mysql_backup_min     = 10,
+  $apt_mirror_hostname,
+  $apt_mirror_gpg_key_fingerprint,
+) {
+  include ::govuk::node::s_base
+  include govuk_env_sync
+
+  file { '/root/.my.cnf':
+    ensure  => 'present',
+    owner   => 'root',
+    group   => 'root',
+    content => template('govuk/node/s_db_admin/.my.cnf.erb'),
```

Comment:
No need for a MySQL equivalent to the [govuk::nodes::postgresql_db_admin class](https://github.com/alphagov/govuk-puppet/blob/f8a80afb0fef55bd2cd61090ae2e7ae8c07e0217/modules/govuk/manifests/nodes/postgresql_db_admin.pp), as the only common config we could extract are the 5-6 lines above, which saves us nothing as we'd have to [pass the constructor arguments](https://github.com/alphagov/govuk-puppet/blob/10871f283d3e8c1c848d89dbba7610b52f5669e6/modules/govuk/manifests/node/s_account_api_db_admin.pp#L35-L41) to the included class.

NB, we're reusing the same `s_db_admin/.my.cnf.erb` file for all, as it contains generic config that will apply to each DB admin machine. The naming is potentially problematic here, as this is no longer being used only by the `s_db_admin` node, but that feels like a refactor we could make later.

Reply:
Yup that sounds reasonable, are you going to change that with a follow up PR? I thought that we could add it as a story to [TODO (Later)](https://trello.com/b/eFUUbwPM/database-migration-dec-2021-jan-2022) but it seemed a bit small. I imagine it could just be: `govuk-puppet/modules/govuk/templates/mysql_my.cnf.erb`

Reply:
I hesitated because I wasn't sure _where_ the shared template should live, but I like your suggestion. I've raised #11356, and will rebase this PR when that's merged.